components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java

```diff
@@ -1597,6 +1600,85 @@ public String getEndpointConfig(String endpointId) throws APIMgtDAOException {
         }
     }
 
+    /**
+     * @see ApiDAO#updateDedicatedGateway(DedicatedGateway, String, Set)
+     */
+    @Override
+    public void updateDedicatedGateway(DedicatedGateway dedicatedGateway, String apiId, Set<String> labels)
+            throws APIMgtDAOException {
+
+        // labels will come in three ways.
+        // 1. auto-generated label
+        // 2. existing label set ( if the updateDedicatedGateway is doing with isEnabled=false for a normal API)
+        // 3. null - when API is updated from dedicatedGatewayIsEnabled=true from dedicatedGatewayIsEnabled=false
+
+        final String query = "UPDATE AM_API SET HAS_OWN_GATEWAY = ?, LAST_UPDATED_TIME = ?, UPDATED_BY = ? " +
+                "WHERE API_ID = ?";
+        try (Connection connection = DAOUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            try {
+
+                statement.setBoolean(1, dedicatedGateway.isEnabled());
+                statement.setTimestamp(2, Timestamp.valueOf(LocalDateTime.now()));
+                statement.setString(3, dedicatedGateway.getUpdatedBy());
+                statement.setString(4, apiId);
+
+                // if the labels are null or perAPI-GW
+                if (labels == null || labels.isEmpty() ||
+                        labels.contains(ContainerBasedGatewayConstants.PER_API_GATEWAY_PREFIX + apiId)) {
+                    deleteLabelsMapping(connection, apiId);
+                    addLabelMapping(connection, apiId, labels);
+
+                }
+                statement.execute();
+                connection.commit();
+            }  catch (SQLException e) {
+                String msg = "Couldn't update dedicated Gateway details of API : " + apiId;
+                connection.rollback();
+                log.error(msg, e);
+                throw new APIMgtDAOException(msg, ExceptionCodes.APIM_DAO_EXCEPTION);
+            } finally {
+                connection.setAutoCommit(DAOUtil.isAutoCommit());
```

Comment:
what the purpose of this line?

Reply:
It is added to the purpose of finally each transaction included in this method to be committed automatically when it is completed. Is it wrong ?